rpc/js/vite.config.ts

```diff
@@ -0,0 +1,42 @@
+/// <reference types="vitest" />
+import path from 'node:path';
+import { defineConfig } from 'vite';
+
+import pkg from './package.json';
+
+// https://vitejs.dev/config/
+export default defineConfig({
+  define: {
+    'process.env.NODE_ENV': '"production"',
+    __VERSION__: JSON.stringify(pkg.version),
+  },
+  build: {
+    commonjsOptions: {
+      transformMixedEsModules: true,
+      include: [
+        /exponential-backoff/u,
+        /google-protobuf/u,
+        /@improbable-eng\/grpc-web/u,
+        /gen\//u,
+      ],
+    },
+    minify: true,
+    target: 'esnext',
```

Comment:
I feel a little weird about changing this value away from [its default](https://vitejs.dev/config/build-options.html#build-target); is it required?

Reply:
```nicolejung@Nicoles-MacBook-Air js % npm run build

> @viamrobotics/rpc@0.2.0 build
> vite build && tsc --emitDeclarationOnly

vite v5.0.6 building for production...
✓ 50 modules transformed.
dist/main.js  660.84 kB │ gzip: 86.49 kB
✓ built in 668ms
nicolejung@Nicoles-MacBook-Air js % npm run build

> @viamrobotics/rpc@0.2.0 build
> vite build && tsc --emitDeclarationOnly

vite v5.0.6 building for production...
✓ 50 modules transformed.
dist/main.js  659.53 kB │ gzip: 86.16 kB
✓ built in 640ms```

The first one was built without the `target` value, while the second one had `target` set to `esnext`. It seems it helped with both speed and space.

Reply:
It's expected that it would be slightly smaller with `esnext`, but I also think it will be less compatible. It may even _require_ further transpilation to actually be usable, which IMO would be undesirable in a published module.

- If we have a concrete reason for deviating from default, we should have a comment explaining why
- If we _don't_ have a concrete reason, we should remove the option to avoid encoding useless superstition

Reply:
Agreed that the default should be fine here